qa/rpc-tests/mempool_size_limit.py

```diff
@@ -0,0 +1,605 @@
+            assert(False)
+
+        prev_mpool = mpool
+        mpool = self.nodes[0].getrawmempool()
+        assert(txid in mpool)
+        assert_equal(len(mpool), len(prev_mpool))
+
+        self.assert_limits_enforced()
+
+        print("Wait for high fee transaction to be present in node1 mempool")
+        timeout = 5
+        waiting = 0
+        mpool1 = self.nodes[1].getrawmempool()
+        while txid not in mpool1 and waiting < timeout:
+            print("Waiting...")
+            waiting += 1
+            time.sleep(1)
+            mpool1 = self.nodes[1].getrawmempool()
+
+        assert(txid in mpool1)
+        assert(len(mpool1) < len(mpool))
+        for tx1 in mpool1:
+            assert(tx1 in mpool)
+            txinfo = self.nodes[1].getrawmempool(True)[tx1]
+            feerate = txinfo['fee'] / txinfo['size']
+            assert(feerate >= minfeerate or tx1 in ctransactions)
+
+        # make sure that chained transactions, known to have aggregated high fee, have not been evicted
+        for ct in ctransactions:
+            assert(ct in mpool)
+            assert(ct in mpool1)
+
+        mark_logs("Sending low fee non ceasing certificates, expecting success", self.nodes, DEBUG_MODE)
+        cert_fees = []
+        nc_certs = [self.nodes[0].sendrawtransaction(certs[2][0]['hex']),
+                    self.nodes[0].sendrawtransaction(certs[3][0]['hex'])]
+        cert_sent_size = (len(certs[2][0]['hex']) + len(certs[3][0]['hex'])) // 2
+        cert_sent = 2
+        cert_fees.append(certs[2][0]['fee'] / (len(certs[2][0]['hex']) // 2))
+        cert_fees.append(certs[3][0]['fee'] / (len(certs[3][0]['hex']) // 2))
+
+        mark_logs("Filling mpool with low fee certificates, expecting success and transaction eviction", self.nodes, DEBUG_MODE)
+        highest_quality_sc0 = 0
+        last_cert_size = 0
+        min_sc0_fee = MAX_FEE
+        while (cert_sent_size < NODE0_CERT_LIMIT_B or usage < NODE0_LIMIT_B - last_cert_size):
+            assert(len(certs[0]) > 0)
+            c = certs[0].pop(0)
+            high_quality_cert_sc0 = self.nodes[0].sendrawtransaction(c['hex'])
+            last_cert_size = len(c['hex']) // 2
+            cert_fees.append(c['fee'] / last_cert_size)
+            if (c['fee'] / last_cert_size < min_sc0_fee):
+                min_sc0_fee = c['fee'] / last_cert_size
+            cert_sent_size += last_cert_size
+            cert_sent += 1
+            usage = int(self.nodes[0].getmempoolinfo()['bytes'])
+
+        mark_logs("Sending one more low fee certificate, expecting failure", self.nodes, DEBUG_MODE)
+        try:
+            c = certs[0].pop(0)
+            self.nodes[0].sendrawtransaction(c['hex'])
+            assert(False)
+        except JSONRPCException as e:
+            assert_equal(e.error['code'], -7)
+            print("Ok")
+
+        mark_logs("Sending high fee certificates, expecting success and sc0 low quality certificate eviction", self.nodes, DEBUG_MODE)
```

Comment:
Similarly to previous comment, it could be worth checking that at least the lowest fee certificate is evicted (I know possibly the eviction affects multiple certificate, but the lowest fee for sure).

Reply:
Agreed to handle this in a separate PR.